### PR TITLE
BP-66: support throttling for zookeeper read during rereplication

### PR DESCRIPTION
**Pull request by thetumbled**
Implementation PR for BP-66.
Master Issue: https://github.com/apache/bookkeeper/issues/4257
Proposal PR: #4258

### CODE REVIEW

**Comment by dlg99** [comment]
Please add the config with description into https://github.com/apache/bookkeeper/blob/master/conf/bk_server.conf and update documentation https://bookkeeper.apache.org/docs/reference/config

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/AbstractConfiguration.java [hunk 1]
```diff
@@ -1236,6 +1237,24 @@ public T setReplicationRateByBytes(int rate) {
         return getThis();
     }
 
+    /**
+     * get the max tasks can be acquired per second of re-replication.
+     * @return max tasks can be acquired per second of re-replication.
+     */
+    public double getReplicationAcquireTaskPerSecond() {
+        return getDouble(REPLICATION_ACQUIRE_TASK_PER_SECOND, 1.0);
```
Comment (dlg99) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/AbstractConfiguration.java:1245]: Let's set default to 0 (no throttling) to maintain existing behavior by default

**Review by dlg99 — CHANGES_REQUESTED** [review]
Please change default to maintain existing behavior and update config/documentation

**Review by shoothzj — COMMENTED** [review]
should `getLedgerToRereplicateFromHierarchy` also beware of this configuration?

**Comment by shoothzj** [comment]
IMO, I think it's a ZooKeeper specified implementation, and maybe we should named this configuration `zk` prefix, like 
```java
    protected static final String ZK_LEDGERS_ROOT_PATH = "zkLedgersRootPath";
    protected static final String ZK_REQUEST_RATE_LIMIT = "zkRequestRateLimit";
```

cc @dlg99 @eolivelli

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/AbstractConfiguration.java (hunk 1 above)
Comment (thetumbled) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/AbstractConfiguration.java]: Good idea, i have changed the implementation, thanks.

**Comment by thetumbled** [comment]
> Please change default to maintain existing behavior and update config/documentation

Fixed, PTAL, thanks.

**Comment by thetumbled** [comment]
> IMO, I think it's a ZooKeeper specified implementation, and maybe we should named this configuration `zk` prefix, like
> 
> ```java
>     protected static final String ZK_LEDGERS_ROOT_PATH = "zkLedgersRootPath";
>     protected static final String ZK_REQUEST_RATE_LIMIT = "zkRequestRateLimit";
> ```
> 
> cc @dlg99 @eolivelli

I have renamed the config.

**Comment by thetumbled** [comment]
> should `getLedgerToRereplicateFromHierarchy` also beware of this configuration?

Not need.

**Review by dlg99 — APPROVED** [review]
Looks good, thank you

**Comment by shoothzj** [comment]
@dlg99 Shall we put this configuration in `Replication Worker settings` section? https://github.com/apache/bookkeeper/blob/09a38e5192bc487ce67c7b041bee7235f79b575c/conf/bk_server.conf#L1051C4-L1051C31

**Comment by dlg99** [comment]
@shoothzj @thetumbled 
> @dlg99 Shall we put this configuration in `Replication Worker settings` section? 

yeah, RW settings seem more relevant

**Comment by thetumbled** [comment]
> @shoothzj @thetumbled
> 
> > @dlg99 Shall we put this configuration in `Replication Worker settings` section?
> 
> yeah, RW settings seem more relevant

I have move it to `Replication Worker settings` section, thanks. @shoothzj @dlg99

**Review by shoothzj — APPROVED** [review]
LGTM

**Comment by thetumbled** [comment]
Please help to review, thanks. @wenbingshen

**Review by wenbingshen — APPROVED** [review]
LGTM

**Comment by shoothzj** [comment]
Thanks for your first contribution, and welcome to the community

**Pull request merged**